test/model/tools/model_test/main.cpp

```diff
@@ -0,0 +1,379 @@
+        UPDATE_SPEC(truncate, float);
+
+        UPDATE_SPEC(checkpoint, float);
+        UPDATE_SPEC(crash, float);
+        UPDATE_SPEC(restart, float);
+        UPDATE_SPEC(set_stable_timestamp, float);
+
+        UPDATE_SPEC(prepared_transaction, float);
+        UPDATE_SPEC(nonprepared_transaction_rollback, float);
+        UPDATE_SPEC(prepared_transaction_rollback_after_prepare, float);
+        UPDATE_SPEC(prepared_transaction_rollback_before_prepare, float);
+
+        else if (k == "connection_config") conn_config += "," + m.get_string("connection_config");
+
+        else if (k == "table_config") table_config += "," + m.get_string("table_config");
+
+        else throw std::runtime_error("Invalid configuration key: " + k);
+    }
+
+#undef UPDATE_SPEC_START
+#undef UPDATE_SPEC
+}
+
+/*
+ * load_spec --
+ *     Load the specification from file. Throw an exception on error.
+ */
+static void
+load_spec(model::kv_workload_generator_spec &spec, std::string &conn_config,
+  std::string &table_config, const char *file)
+{
+    std::ifstream f(file);
+    if (!f.is_open())
+        throw std::runtime_error("Cannot open file: " + std::string(file));
+
+    std::string line;
+    while (std::getline(f, line)) {
+        line = trim(line);
+        if (line.empty() || line[0] == '#')
+            continue;
+        update_spec(spec, conn_config, table_config, line.c_str());
+    }
+    if (!f.eof() && !f.good())
+        throw std::runtime_error("Cannot read from file: " + std::string(file));
+
+    f.close();
+}
+
+/*
+ * usage --
+ *     Print usage help for the program. (Don't exit.)
+ */
+static void
+usage(const char *progname)
+{
+    fprintf(stderr, "usage: %s [OPTIONS]\n\n", progname);
+    fprintf(stderr, "Options:\n");
+    fprintf(stderr, "  -C CONFIG  specify WiredTiger's connection configuration\n");
+    fprintf(stderr, "  -G CONFIG  specify the workload generator's configuration\n");
+    fprintf(stderr, "  -h HOME    specify the database directory\n");
+    fprintf(stderr, "  -I n       run the test for at least this many iterations\n");
+    fprintf(stderr, "  -i FILE    load the generator's configuration from the file\n");
+    fprintf(stderr, "  -l N[-M]   specify the workload length as a number of transactions\n");
+    fprintf(stderr, "  -M N[-M]   specify the number of tables\n");
+    fprintf(stderr, "  -n         do not execute the workload; only print it\n");
+    fprintf(stderr, "  -p         preserve the last database directory\n");
+    fprintf(stderr, "  -S SEED    specify the random number generator's seed\n");
+    fprintf(stderr, "  -T CONFIG  specify WiredTiger's table configuration\n");
+    fprintf(stderr, "  -t N       repeat the test for at least this number of seconds\n");
+    fprintf(stderr, "  -?         show this message\n");
+}
+
+/*
+ * main --
+ *     The main entry point for the test.
+ */
+int
+main(int argc, char *argv[])
+{
+    model::kv_workload_generator_spec spec;
```

Comment:
I can't remember sorry - what does the style guide say about declaring before use? It feels a bit outdated/pointless in C++-land, but I can understand if it's required for some consistency with the C parts of the codebase.

Reply:
Fixed; there is no reason that all these variables should be declared before initialization. A power of habit.